src/main/kotlin/com/exactpro/th2/read/file/common/impl/LineParser.kt

```diff
@@ -28,10 +29,10 @@ import java.nio.charset.MalformedInputException
 import java.util.function.BiPredicate
 import java.util.function.Function
 
-open class LineParser @JvmOverloads constructor(
+abstract class LineParser<MESSAGE_BUILDER> @JvmOverloads constructor(
     private val filter: BiPredicate<StreamId, String> = BiPredicate { _, _ -> true },
     private val transformer: Function<String, String> = Function { it }
-) : ContentParser<BufferedReader> {
+) : ContentParser<BufferedReader, MESSAGE_BUILDER> {
```

Comment:
I think we do that instead

```kotlin
open class LineParser<MESSAGE_BUILDER> @JvmOverloads constructor(
    private val filter: BiPredicate<StreamId, String> = BiPredicate { _, _ -> true },
    private val transformer: Function<String, String> = Function { it },
    private val lineToBuilder: (String, Charset) -> MESSAGE_BUILDER,
) : ContentParser<BufferedReader, MESSAGE_BUILDER> {
  //....
  companion object {
    @JvmField
    val PROTO: (String, Charset) -> ProtoRawMessage.Builder = { TODO() }
    val TRANSPORT: (String, Charset) -> RawMessage.Builder = { TODO() }
  }
}
```

I don't really want to see several classes instead of a single one if we can solve the problem in a different way

Reply:
In the same way, you instantiate the correct implementation right now... I don't see any problems with that...
The only request here is to avoid creating a class hierarchy where it is not really necessary. I my opinion the aggregation should work better in this case (for readability as well as for the end user)

Reply:
@lumber1000 Do you have any updates on this?